src/helics/core/Core.hpp

```diff
@@ -810,14 +810,34 @@ class Core {
      * @param tag a string containing the name of the tag
      * @param value a string containing the value for the tag
      */
-    virtual void
-        setTag(InterfaceHandle handle, const std::string& tag, const std::string& value) = 0;
+    virtual void setInterfaceTag(InterfaceHandle handle,
+                                 const std::string& tag,
+                                 const std::string& value) = 0;
     /**
      * getter for the interface tags
      * @param handle the identifier for the interface to set the info data on
      * @param tag the name of the tag to retrieve
      */
-    virtual const std::string& getTag(InterfaceHandle handle, const std::string& tag) const = 0;
+    virtual const std::string& getInterfaceTag(InterfaceHandle handle,
+                                               const std::string& tag) const = 0;
+
+    /**
+     * setter for federate tags which are key-value pairs
+     * @param fid the identifier for the federate to set the tag data on
+     * @param tag a string containing the name of the tag
+     * @param value a string containing the value for the tag
+     */
+    virtual void
+        setFederateTag(LocalFederateId fid, const std::string& tag, const std::string& value) = 0;
+    /**
+     * getter for the federate tags
+     * @details this returns a std::string since the string that may be returned is not guaranteed
```

Comment:
Should the function be returning an `std::string` instead of an `std::string&`?

Reply:
I actually had it as a std::string for a while, but that made the C interface problematic so I changed it to a const std::string  &.  Guess I forgot to change the documentation on the method.  I will need to fix that.